shopify/api_access.py

```diff
@@ -0,0 +1,40 @@
+import re
+
+
+class ApiAccess:
+
+    SCOPE_DELIMITER = ","
+    IMPLIED_SCOPE_RE = re.compile(r"\A(?P<unauthenticated>unauthenticated_)?write_(?P<resource>.*)\Z")
+
+    def __init__(self, scopes):
+
+        if type(scopes) == str:
+            scopes = scopes.split(self.SCOPE_DELIMITER)
+
+        self.__store_scopes(scopes)
+
+    def covers(self, api_access):
+        return api_access._compressed_scopes <= self._expanded_scopes
+
+    def __str__(self):
+        return self.SCOPE_DELIMITER.join(self._compressed_scopes)
+
+    def __iter__(self):
+        return iter(self._compressed_scopes)
+
+    def __eq__(self, other):
+        return type(self) == type(other) and self._compressed_scopes == other._compressed_scopes
+
+    def __store_scopes(self, scopes):
+        sanitized_scopes = set(list(filter(None, [scope.strip() for scope in scopes])))
```

Comment:
If we don't want/need the contents of the set to change after we make it, we could make this a `frozenset`.

Also, I don't think you need to convert to `list` before converting to `set`/`frozenset`.